readme.md
## WebMan Toggle Admin Bar

Toggles the WordPress admin bar the right way. Theme developer's best friend!

## Description

**[WebMan Toggle Admin Bar](http://www.webmandesign.eu/ "WebMan Design") is WordPress plugin that provides an easy way to toggle the WordPress admin bar on your website's front end.**

This plugin actually toggles the whole admin bar functionality, so it gets rid of CSS styles and `.admin-bar` class that the WordPress admin bar applies. Thus, the plugin is perfectly suited for WordPress theme developers to test their themes with and without the admin bar.

Please note that once you click the plugin's button in the bottom left corner of the screen, the page will be reloaded. The plugin uses no JavaScript.

### Plugin Localization

Have a translation? Please post it on the [support forum](http://support.webmandesign.eu/ "WebMan Support Forum").

Please, find the instructions on how to localize the plugin in `webman-toggle-admin-bar/languages/readme.md` file.

### Additional Resources

* [Write a review](http://wordpress.org/support/view/plugin-reviews/webman-toggle-admin-bar#postform)
* [Have a question?](https://wordpress.org/support/plugin/webman-toggle-admin-bar)
* [Follow @webmandesigneu](https://twitter.com/webmandesigneu)
* [Visit WebMan Design](http://www.webmandesign.eu)

## Installation

1. Unzip the plugin download file and upload `webman-toggle-admin-bar` folder into the `/wp-content/plugins/` directory.
2. Activate the plugin through the *"Plugins"* menu in WordPress.
3. Once you you click an admin bar toggle button in the bottom left corner of your website's front end, the admin bar will toggle.

## Frequently Asked Questions

### How to translate (localize) the plugin?

Please, find the instructions on how to localize the plugin in `webman-amplifier/languages/readme.md` file.

## Changelog

Please see the [`changelog.md` file](https://github.com/webmandesign/webman-toggle-admin-bar/blob/master/changelog.md) for details.

changelog.md
# WebMan Toggle Admin Bar Changelog

## 1.2

* Update: Improved code
* Update: Hardening security

#### Files changed:

	includes/class-webman-toggle-admin-bar.php


## 1.1

* Fixed: PHP error in theme customizer

#### Files changed:

	includes/class-webman-toggle-admin-bar.php


## 1.0

* Initial release.

webman-toggle-admin-bar.php
<?php

/**
 * WebMan Toggle Admin Bar
 *
 * A plugin for easy testing WordPress themes with and without admin bar.
 * The admin bar needs to be enabled.
 *
 * @package    WebMan Toggle Admin Bar
 * @copyright  2015 WebMan - Oliver Juhas
 * @license    GPL-2.0+, http://www.gnu.org/licenses/gpl-2.0.html
 *
 * @wordpress-plugin
 * Plugin Name:        WebMan Toggle Admin Bar
 * Plugin URI:         http://www.webmandesign.eu/
 * Description:        A plugin for disabling and enabling WordPress admin bar on front end. The admin bar needs to be enabled in WordPress admin first.
 * Version:            1.2
 * Author:             WebMan - Oliver Juhas
 * Author URI:         http://www.webmandesign.eu/
 * License:            GPL-2.0+
 * License URI:        http://www.gnu.org/licenses/gpl-2.0.txt
 * Text Domain:        wm_domain
 * Domain Path:        /languages
 * Requires at least:  4.1
 * Tested up to:       4.2
 */



//Exit if accessed directly
if ( ! defined( 'WPINC' ) ) exit;





/**
 * Constants
 */

	//Define global constants

		//Paths
			if ( ! defined( 'WMTAB_PLUGIN_FILE' ) ) define( 'WMTAB_PLUGIN_FILE', __FILE__ );
			if ( ! defined( 'WMTAB_PLUGIN_DIR' ) )  define( 'WMTAB_PLUGIN_DIR',  trailingslashit( plugin_dir_path( __FILE__ ) ) );
			if ( ! defined( 'WMTAB_PLUGIN_URL' ) )  define( 'WMTAB_PLUGIN_URL',  trailingslashit( plugin_dir_url( __FILE__ ) ) );





/**
 * Plugin setup
 */

	/**
	 * Plugin activation
	 *
	 * @since    1.0
	 * @version  1.0
	 */
	function wmtab_activate() {
		do_action( 'wmhook_wmtab_plugin_activation' );
	} // /wmtab_activate

	register_activation_hook( WMTAB_PLUGIN_FILE, 'wmtab_activate' );



	/**
	 * Plugin deactivation
	 *
	 * @since    1.0
	 * @version  1.0
	 */
	function wmtab_deactivate() {
		do_action( 'wmhook_wmtab_plugin_deactivation' );
	} // /wmtab_deactivate

	register_deactivation_hook( WMTAB_PLUGIN_FILE, 'wmtab_deactivate' );



	/**
	 * Load the plugin
	 */

		/**
		 * Load the plugin main class
		 */
		require_once( WMTAB_PLUGIN_DIR . 'includes/class-webman-toggle-admin-bar.php' );



		/**
		 * Execute the plugin
		 *
		 * @since    1.0
		 * @version  1.0
		 *
		 * @return  WebMan Toggle Admin Bar instance
		 */
		function wm_toggle_admin_bar() {
			$plugin = new WM_Toggle_Admin_Bar();
			$plugin->run();
		} // /wm_toggle_admin_bar

		wm_toggle_admin_bar();


includes/class-webman-toggle-admin-bar.php
<?php
/**
 * WebMan Toggle Admin Bar
 *
 * @package  WebMan Toggle Admin Bar
 *
 * @link  http://www.webmandesign.eu
 *
 * @since    1.0
 * @version  1.2
 */



//Exit if accessed directly
if ( ! defined( 'WPINC' ) ) exit;





/**
 * Main WM Toggle Admin Bar class
 *
 * Contains the main functions for WebMan Toggle Admin Bar.
 *
 * @since    1.0
 * @version	 1.0
 */
if ( ! class_exists( 'WM_Toggle_Admin_Bar' ) ) {

	class WM_Toggle_Admin_Bar {

		/**
		 * Run the plugin
		 */

			/**
			 * Constructor
			 *
			 * A dummy constructor to prevent class from being loaded more than once.
			 *
			 * @since    1.0
			 * @version  1.0
			 */
			public function __construct() {

				/* Do nothing here */

			} // /__construct



			/**
			 * Run the loader to execute all of the hooks with WordPress.
			 *
			 * @since    1.0
			 * @version  1.0
			 */
			public function run() {

				//Store the instance locally to avoid private static replication
					static $instance = null;

				//Only run these methods if they haven't been ran previously
					if ( null === $instance ) {
						$instance = new WM_Toggle_Admin_Bar;

						$instance->setup_actions();
					}

				//Always return the instance
					return $instance;

			} // /run





		/**
		 * Methods
		 */

			/**
			 * Setup the default action hooks
			 *
			 * @since    1.0
			 * @version  1.0
			 */
			public function setup_actions() {

				add_action( 'plugins_loaded',     array( $this, 'load_textdomain'  ) );
				add_action( 'after_setup_theme',  array( $this, 'toggle_admin_bar' ) );
				add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_assets'   ) );
				add_action( 'wp_footer',          array( $this, 'toggle_button'    ), 9998 );

			} // /setup_actions



			/**
			 * Can the button be displayed?
			 *
			 * @since    1.0
			 * @version  1.1
			 */
			public function can_display_button() {

				if (
						! is_user_logged_in()
						|| is_admin()
						|| isset( $GLOBALS['wp_customize'] )
					) {
					return false;
				}

				return true;

			} // /can_display_button



			/**
			 * Enqueue styles and scripts
			 *
			 * @since    1.0
			 * @version  1.1
			 */
			public function enqueue_assets() {

				//Requirements check
					if ( ! $this->can_display_button() ) {
						return;
					}

				//Helper variables
					$plugin_data = ( function_exists( 'get_plugin_data' ) ) ? ( get_plugin_data( WMTAB_PLUGIN_FILE, false ) ) : ( array( 'Version' => $GLOBALS['wp_version'] ) );

				//Enqueue assets
					wp_enqueue_style(
							'wmtab-styles',
							WMTAB_PLUGIN_URL . 'assets/css/style.css',
							false,
							$plugin_data['Version'],
							'screen'
						);

			} // /enqueue_assets



			/**
			 * Toggle button
			 *
			 * @link  https://kovshenin.com/2012/current-url-in-wordpress/
			 *
			 * @since    1.0
			 * @version  1.2
			 */
			public function toggle_button() {

				//Requirements check
					if ( ! $this->can_display_button() ) {
						return;
					}

				//Helper variables
					global $wp;

					$url = ( isset( $wp->request ) ) ? ( add_query_arg( $wp->query_string, '', home_url( $wp->request ) ) ) : ( false );

					//Requirements check
						if ( ! $url ) {
							return;
						}

						$url = esc_url( remove_query_arg( array( 'wmtab-on', 'wmtab-off' ), $url ) );
						$url = esc_url( add_query_arg( 'wmtab-on', '', $url ) );

				//Processing
					if ( ! is_admin_bar_showing() ) {

						$url = str_replace( 'wmtab-off', 'wmtab-on', $url );

						$button_text = _x( 'On', 'Button label. Enable admin bar.', 'wm_domain' );

					} else {

						$url = str_replace( 'wmtab-on', 'wmtab-off', $url );

						$button_text = _x( 'Off', 'Button label. Disable admin bar.', 'wm_domain' );

					}

				//Output
					echo '<div class="wm-toggle-admin-bar-container"><a href="' . esc_url( $url ) . '" title="' . __( 'Toggle admin bar', 'wm_domain' ) . '" class="wm-toggle-admin-bar">' . $button_text . '</a></div>';

			} // /toggle_button



			/**
			 * Toggle admin bar
			 *
			 * @since    1.0
			 * @version  1.0
			 */
			public function toggle_admin_bar() {

				//Requirements check
					if ( ! $this->can_display_button() ) {
						return;
					}

				//Toggle admin bar
					if (
							isset( $_GET['wmtab-on'] )
							&& ! is_admin_bar_showing()
						) {

						add_filter( 'show_admin_bar', '__return_true' );

					} elseif (
							isset( $_GET['wmtab-off'] )
							&& is_admin_bar_showing()
						) {

						add_filter( 'show_admin_bar', '__return_false' );

					}

			} // /toggle_admin_bar





		/**
		 * Localization
		 */

			/**
			 * Localization
			 *
			 * Load the translation file for the current language.
			 *
			 * Checks the languages folder inside the plugin first,
			 * and then the default WordPress languages folder.
			 *
			 * Note: the first-loaded translation file overrides any
			 * following ones if the same translation is present.
			 *
			 * @since    1.0
			 * @version  1.0
			 */
			public function load_textdomain() {

				//Traditional WordPress plugin locale filter
					$locale = apply_filters( 'plugin_locale', get_locale(), 'wm_domain' );
					$mofile = $locale . '.mo';

				//Look in local /wp-content/plugins/webman-toggle-admin-bar/languages/ folder
					load_textdomain( 'wm_domain', trailingslashit( WMTAB_PLUGIN_DIR ) . 'languages/' . $mofile );

				//Look in global /wp-content/languages/webman-toggle-admin-bar folder
					load_textdomain( 'wm_domain', trailingslashit( WP_LANG_DIR ) . 'plugins/webman-toggle-admin-bar/' . $mofile );

			} // /load_textdomain

	} // /WM_Toggle_Admin_Bar

} // /class WM_Toggle_Admin_Bar check
